### PR TITLE
Dlframe examples of image model inference and image transfer learning

### PR DESCRIPTION
**Pull request by songhappy**
## What changes were proposed in this pull request?

add 
example.DLframes.imageInference.ImageInference and its document, example.DLframes.imageTransferLearning.ImageTransferLearning and its document

## How was this patch tested?
manual tested on local and cluster

## Related links or issues (optional)
raised  https://github.com/intel-analytics/BigDL/issues/2392

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/ImageTransferLearning.scala [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      val evaluation = new MulticlassClassificationEvaluator().setPredictionCol("prediction")
+        .setMetricName("accuracy").evaluate(predictions)
+      println("evaluation result on validationDF: " + evaluation)
+
+    }
+  }
+
+}
+
+class Array2Vector(override val uid: String = "array2vector") extends Transformer with HasFeaturesCol
```
Comment (hhbyyh) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/ImageTransferLearning.scala]: can we use a udf for this?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageInference/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+## Overview
+   Deep Learning Frames provides high-level APIs for scalable deep learning in Scala with Apache Spark.
+   The current version of Deep Learning Frames provides a suite of tools around working with and processing images using deep learning. 
+   Two exmaples demostrates how to use BigDL to apply popular iamge deep learning models at scale.
```
Comment (hhbyyh) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageInference/README.md]: the readme can focus on one example only.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+## Overview
+   Deep Learning Frames provides high-level APIs for scalable deep learning in Scala with Apache Spark.
+   The current version of Deep Learning Frames provides a suite of tools around working with and processing images using deep learning. 
+   this exmaple demostrates how to use BigDL for transfer learning.
+
+## Transfer Learning 
+   1. DLFrames provides utilities to perform transfer learning on images, which is one of the fastest (code and run-time-wise) ways to start using deep learning.             
+         
+          val createLabel = udf((name: String) => if (name.contains("cat")) 1.0 else 0.0)
```
Comment (hhbyyh) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md]: We already got the example code, so I'm thinking either remove code here or only leave the most critical lines.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          val pipeline = new Pipeline().setStages(
+            Array(featurizer, schemaTransformer, lr))
+    
+          val pipelineModel = pipeline.fit(trainingDF)
+    
+          val predictions = pipelineModel.transform(trainingDF)
+         
+   2. You can run the full TransferLearning example by following steps.
+        
+        2.1 Prepare pre-trained model and defenition file as Model Inference.
```
Comment (hhbyyh) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md]: Please add the reference for the model used in this example to help users repro it.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/ImageTransferLearning.scala (hunk 1 above)
Comment (songhappy) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/ImageTransferLearning.scala]: I want to use the pipeline then I have to do this this way. If I want to manually connect the steps, this is not necessary

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageInference/README.md (hunk 2 above)
Comment (songhappy) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageInference/README.md]: ok

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md (hunk 3 above)
Comment (songhappy) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md]: will delete a couple of lines

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md (hunk 4 above)
Comment (songhappy) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/dlframes/imageTransferLearning/README.md]: good suggestion

**Comment by hhbyyh** [comment]
Thanks for working on this.
Next time please wait to peer review before merging it. 
Also please run PR validation before merging to master.
I'll change the pre-processing part.

**Pull request merged**